test/extensions/filters/http/cache/http_cache_implementation_test_common.cc

```diff
@@ -0,0 +1,732 @@
+    response_headers_ptr = move(lookup_result.headers_);
+  });
+  return response_headers_ptr;
+}
+
+std::string HttpCacheImplementationTest::getBody(LookupContext& context, uint64_t start,
+                                                 uint64_t end) {
+  AdjustedByteRange range(start, end);
+  std::string body;
+  context.getBody(range, [&body](Buffer::InstancePtr&& data) {
+    EXPECT_NE(data, nullptr);
+    if (data) {
+      body = data->toString();
+    }
+  });
+  return body;
+}
+
+Http::TestResponseTrailerMapImpl HttpCacheImplementationTest::getTrailers(LookupContext& context) {
+  Http::TestResponseTrailerMapImpl trailers;
+  context.getTrailers([&trailers](Http::ResponseTrailerMapPtr&& data) {
+    if (data) {
+      trailers = *data;
+    }
+  });
+  return trailers;
+}
+
+LookupRequest HttpCacheImplementationTest::makeLookupRequest(absl::string_view request_path) {
+  request_headers_.setPath(request_path);
+  return LookupRequest(request_headers_, time_source_.systemTime(), vary_allow_list_);
+}
+
+testing::AssertionResult HttpCacheImplementationTest::expectLookupSuccessWithHeaders(
+    LookupContext* lookup_context, const Http::TestResponseHeaderMapImpl& headers) {
+  if (lookup_result_.cache_entry_status_ != CacheEntryStatus::Ok) {
+    return AssertionFailure() << "Expected: lookup_result_.cache_entry_status == "
+                                 "CacheEntryStatus::Ok\n  Actual: "
+                              << lookup_result_.cache_entry_status_;
+  }
+  if (!lookup_result_.headers_) {
+    return AssertionFailure() << "Expected nonnull lookup_result_.headers";
+  }
+  if (!lookup_context) {
+    return AssertionFailure() << "Expected nonnull lookup_context";
+  }
+
+  Http::ResponseHeaderMapPtr actual_headers_ptr = getHeaders(*lookup_context);
+  if (!TestUtility::headerMapEqualIgnoreOrder(headers, *actual_headers_ptr)) {
+    return AssertionFailure() << "Expected headers: " << headers
+                              << "\nActual:  " << *actual_headers_ptr;
+  }
+  return AssertionSuccess();
+}
+
+testing::AssertionResult HttpCacheImplementationTest::expectLookupSuccessWithBodyAndTrailers(
+    LookupContext* lookup_context, absl::string_view body,
+    Http::TestResponseTrailerMapImpl trailers) {
+  if (lookup_result_.cache_entry_status_ != CacheEntryStatus::Ok) {
+    return AssertionFailure() << "Expected: lookup_result_.cache_entry_status == "
+                                 "CacheEntryStatus::Ok\n  Actual: "
+                              << lookup_result_.cache_entry_status_;
+  }
+  if (!lookup_result_.headers_) {
+    return AssertionFailure() << "Expected nonnull lookup_result_.headers";
+  }
+  if (!lookup_context) {
+    return AssertionFailure() << "Expected nonnull lookup_context";
+  }
+  const std::string actual_body = getBody(*lookup_context, 0, body.size());
+  if (body != actual_body) {
+    return AssertionFailure() << "Expected body == " << body << "\n  Actual:  " << actual_body;
```

Comment:
Optional: Perhaps absl::StrFormat() is easier to read in a bunch of these concatenated strings?

Reply:
I can't find a way to customize how absl::StrFormat() or absl::Substitute() print values to strings, so we'd have to add string converstion boilerplate which probably balances out the readability gains.